airflow/contrib/hooks/elasticsearch_hook.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow.hooks.base_hook import BaseHook
+from airflow import configuration
+
+from elasticsearch import Elasticsearch, ElasticsearchException
+from airflow.exceptions import AirflowException
+
+class ElasticsearchHook(BaseHook, Elasticsearch):
```

Comment:
hey thanks a lot for spotting this. I actually harbor this question for a long time: are hooks supposed to house the method that we use? or are hooks supposed to just assist we generate whatever client we need by getting parameters via Connections? 

From the documentation and code I have read so far, I'm observing two different perspectively. For example: 

- From the [airflow documentation](https://airflow.incubator.apache.org/concepts.html), we have: 

> Hooks are interfaces to external platforms and databases like Hive, S3, MySQL, Postgres, HDFS, and Pig. 

This means that the methods should be contained within the hook instance. We should follow the multi-inheritance implementation in this PR. 

- However, the implementation of [S3 hook](https://github.com/apache/incubator-airflow/blob/master/airflow/hooks/S3_hook.py) is more of a mixture. We do have `get_conn` meethod, while having some house keeping methods like `check_for_key`. 

- Finally, `get_conn` method might not serve us well if we have long running connections like RabbitMQ. For RabbitMQ, we might wanna establish a connection and only terminate it when all transactions are done. 

In sum the question I wanna ask is: what is a clear guideline on when to have the method persisted within the hook, and when should we separate out to another function?